src/main/java/com/synopsys/integration/detect/lifecycle/autonomous/AutonomousManager.java

```diff
@@ -144,10 +142,35 @@ private Optional<String> findScanType(List<String> adoptedScanTypes, String key)
         }).findFirst();
     }
 
+    private Optional<String> findDetectorType(List<String> detectorTypes, String key) {
+        return detectorTypes.stream().filter(detector -> {
+            String tempKey = key.replace(".","_");
+            return tempKey.contains(detector.toLowerCase());
+        }).findFirst();
+    }
+
+    private void updateScanOrDetectorProperty(String propertyType, Boolean userProvidedProperty, String propertyTypeName, String propertyKey, String propertyValue) {
```

Comment:
Combined the whole function as one in order to avoid repetition.

Reply:
`Boolean userProvidedProperty` can be `boolean userProvidedProperty` here (I see argument values are of type `boolean` already)